src/components/Calculator.js

```diff
@@ -1,30 +1,41 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 export default function Calculator() {
-  const result = 0;
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (buttonName) => {
+    const newState = calculate(state, buttonName);
+    setState(newState);
+  };
+
   return (
     <div className="container">
-      <div className="result">{result}</div>
-      <button type="button" className="btn-light btn-zero">0</button>
-      <button type="button" className="btn-light btn-dot">.</button>
-      <button type="button" className="btn-coloured btn-equal">=</button>
-      <button type="button" className="btn-light btn-clear">AC</button>
-      <button type="button" className="btn-light btn-sign">+/-</button>
-      <button type="button" className="btn-light btn-percentage">%</button>
-      <button type="button" className="btn-coloured btn-divider">/</button>
-      <button type="button" className="btn-coloured btn-multiply">*</button>
-      <button type="button" className="btn-coloured btn-substract">-</button>
-      <button type="button" className="btn-coloured btn-plus">+</button>
-      <button type="button" className="btn-light btn-one">1</button>
-      <button type="button" className="btn-light btn-two">2</button>
-      <button type="button" className="btn-light btn-three">3</button>
-      <button type="button" className="btn-light btn-four">4</button>
-      <button type="button" className="btn-light btn-five">5</button>
-      <button type="button" className="btn-light btn-six">6</button>
-      <button type="button" className="btn-light btn-seven">7</button>
-      <button type="button" className="btn-light btn-eight">8</button>
-      <button type="button" className="btn-light btn-nine">9</button>
+      <div className="result">{state.next || state.total || '0'}</div>
+      <button type="button" className="btn-light btn-zero" onClick={() => handleClick('0')}>0</button>
+      <button type="button" className="btn-light btn-dot" onClick={() => handleClick('.')}>.</button>
+      <button type="button" className="btn-coloured btn-equal" onClick={() => handleClick('=')}>=</button>
+      <button type="button" className="btn-light btn-clear" onClick={() => handleClick('AC')}>AC</button>
```

Comment:
- [ OPTIONAL ] I highly recommend while adding the operation in the calucluator to let the user be able to see what is the operation is doing? so kindly add it, it will improve from user ecperience 🤩🔥💯🙏🏻

![image](https://user-images.githubusercontent.com/58553711/227396897-a70c1c5e-09fc-48e3-a0ae-89cad502cf29.png)